File: WiggleSortII.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    void wiggleSort(vector<int>& nums) {
    	
    	vector<int> temp = nums;
    	
    	sort(temp.begin(), temp.end());
    	
    	int n = nums.size();
    	int i = (n + 1) / 2, j = n;
    	for(int k = 0; k < n; k++) {
    		nums[k] = k & 1 ? temp[--j] : temp[--i];
		}
    	
        
    }
};

int main() {
	
	int n;
	cin>>n;
	
	vector<int> nums;
	for(int i = 0; i < n; i++) {
		int num;
		cin>>num;
		nums.push_back(num);
	}
	
	Solution *solution = new Solution();
	solution->wiggleSort(nums);
	
	for(int i = 0; i < n; i++) cout<<nums[i]<<"  ";
	
	return 0;
	
}

File: WiggleSortII.py
class Solution:
    def wiggleSort(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        temp = sorted(nums)
        n = len(nums)
        r1 = (n + 1) // 2 - 1
        r2 = n - 1
        for i in range(n):
            if i & 1:
                nums[i] = temp[r2]
                r2 -= 1
            else:
                nums[i] = temp[r1]
                r1 -= 1


if __name__ == '__main__':
    nums = [int(num) for num in input().split()]
    solution = Solution()
    solution.wiggleSort(nums)
    print(nums)
